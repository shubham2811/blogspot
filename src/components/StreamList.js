import React from 'react';
const StreamList=()=>{
    return (
        <div>
            Route to Stream List
        </div>
    )
}

export default StreamList