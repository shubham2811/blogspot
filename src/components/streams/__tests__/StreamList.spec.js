import { shallow} from 'enzyme';
import React from 'react';
import { Provider } from 'react-redux'
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import { streams } from '../../../apis/streams';
import moxios from 'moxios';
import thunk from 'redux-thunk';
import configureStore from 'redux-mock-store';
import StreamList from '../StreamList';
import * as actions from '../../../actions/index';
import * as types from '../../../actions/types';

const mockStore = configureStore([thunk])
const initialState = {
    "streams":[],
    "auth":{}
   
}
describe('Stream StreamList test with Enzyme', () => {
const store = mockStore(initialState)
  const wrapper = shallow(<StreamList store={store} /> )  

    it('---1---render stream List component initially', () => {
        // expect(wrapper).toEqual({})
        expect(toJson(wrapper)).toMatchSnapshot()
     });


      xit('--2-- dispatching the actions', () => {
        const streams =  [
            {
                "title": "a",
                "description": "a",
                "userId": "118440351538142376154",
                "id": 1
              }
          ]
          const expectedActions = {
            type: types.FETCH_STREAMS,
            payload: streams
          }

          const getActions = store.getActions()
          return store.dispatch(actions.fetchStreams()).then(() => {
            // return of async actions
            expect(getActions[0]).toEqual(expectedActions)
          })
      })
})