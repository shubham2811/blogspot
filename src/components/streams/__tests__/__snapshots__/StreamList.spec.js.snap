// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stream StreamList test with Enzyme ---1---render stream List component initially 1`] = `
<StreamList
  fetchStreams={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": [Function],
    }
  }
  streams={Array []}
/>
`;
